test/cflat-tests.ts

```diff
@@ -0,0 +1,137 @@
+            compiler.filterUserOptions(['-d', 'something', 'something-else']).should.deep.equal([]);
+        });
+
+        it('CflatCompiler should not allow -s parameter', () => {
+            const compiler = new CflatCompiler(info, env);
+            compiler.filterUserOptions(['-s', '--something', '--something-else']).should.deep.equal([]);
+            compiler.filterUserOptions(['-s', 'something', 'something-else']).should.deep.equal([]);
+        });
+
+        it('CflatCompiler should not allow --source-path parameter', () => {
+            const compiler = new CflatCompiler(info, env);
+            compiler.filterUserOptions(['-source-path', '--something', '--something-else']).should.deep.equal([]);
+            compiler.filterUserOptions(['-source-path', 'something', 'something-else']).should.deep.equal([]);
+        });
+
+        it('CflatCompiler should not allow -sourcepath parameter', () => {
+            const compiler = new CflatCompiler(info, env);
+            compiler.filterUserOptions(['-sourcepath', '--something', '--something-else']).should.deep.equal([]);
+            compiler.filterUserOptions(['-sourcepath', 'something', 'something-else']).should.deep.equal([]);
+        });
+    });
+});
+
+describe('cflatp parsing', () => {
+    let compiler: CflatCompiler;
+    let env: CompilationEnvironment;
+    before(() => {
+        env = makeCompilationEnvironment({languages});
+        compiler = new CflatCompiler(info, env);
+    });
+
+    async function testCflat(baseFolder: string, ...classNames: string[]) {
+        const compiler = new CflatCompiler(info, env);
+
+        const asm = classNames.map(className => {
+            return {text: fs.readFileSync(`${baseFolder}/${className}.asm`).toString()};
+        });
+
+        const output = utils.splitLines(fs.readFileSync(`${baseFolder}/output.asm`).toString());
+        const expectedSegments = output.map(line => {
+            const match = line.match(/^line (\d+):(.*)$/);
+            if (match) {
+                return {
+                    text: match[2],
+                    source: {
+                        line: parseInt(match[1]),
+                        file: null,
+                    },
+                };
+            }
+            return {
+                text: line,
+                source: null,
+            };
+        });
+
+        const result = {
+            asm,
+        };
+
+        const processed = await compiler.processAsm(result, this.filters, this.options);
+        processed.should.have.property('asm');
+        const asmSegments = (processed as {asm: ParsedAsmResultLine[]}).asm;
+        asmSegments.should.deep.equal(expectedSegments);
+    }
+
+    it('should handle errors', async () => {
+        const result = {
+            asm: '<Compilation failed>',
+        };
+        (await compiler.processAsm(result, this.filters, this.options)).should.deep.equal([
```

Comment:
There is an error here: "filters" and "options" are undefined. How can we call them? @maemre

Reply:
I left some comments also on the commit. I don't understand what you're trying to test here. What's the point of calling `processAsm`? You can log the `this` object to see which properties it has.

Reply:
We couldn't see your comments on the commit, where are they? 
Also, we call processAsm because we followed the way that Java test handles error. The purpose of this code is likely to test that, when given certain inputs (result, this.filters, and this.options), the processAsm function on the compiler object returns an error message in the expected format. That's how we decide to handle the case of `it('should handle errors', async () => {`.
Since eventaully the expected output is an array with a single object with the properties text equal to '<Compilation failed>' and source equal to null, can we just replace `this.filters` and `this.options` with `null`?

Reply:
> We couldn't see your comments on the commit, where are they?

They also appear on the PR. You can click on the particular commit and load the relevant files to see the comments.

I don't think you even need a `processAsm` function, you don't need to test it if you aren't overloading it. You just need to test that the resulting output file is what the compiler produces.